sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -267,4 +267,50 @@ inline bool operator>=(std::chrono::system_clock::time_point const& tp, DateTime
   return (dt <= tp);
 }
 
+namespace _internal {
+  /**
+   * @brief Provides convertion methods for POSIX time to an #Azure::DateTime.
+   *
+   */
+  class PosixTimeConverter final {
+  public:
+    /**
+     * @brief Converts POSIX time to an #Azure::DateTime.
+     *
+     * @param posixTime The number of seconds since 1970.
+     * @return Calculated #Azure::DateTime.
+     */
+    static DateTime PosixTimeToDateTime(int64_t posixTime)
+    {
+      return DateTime(1970) + std::chrono::seconds(posixTime);
+    }
+
+    /**
+     * @brief Converts a DateTime to POSIX time.
+     *
+     * @param dateTime The `%DateTime` to convert.
+     * @return The number of seconds since 1970.
+     */
+    static int64_t DateTimeToPosixTime(DateTime const& dateTime)
+    {
+      //  This count starts at the POSIX epoch which is January 1st, 1970 UTC.
+      return std::chrono::duration_cast<std::chrono::seconds>(dateTime - DateTime(1970)).count();
+    }
+
+  private:
+    /**
+     * @brief An instance of `%PosixTimeConverter` class cannot be created.
+     *
+     */
+    PosixTimeConverter() = delete;
+
+    /**
+     * @brief An instance of `%PosixTimeConverter` class cannot be destructed, because no instance
+     * can be created.
+     *
+     */
+    ~PosixTimeConverter() = delete;
```

Comment:
Other than the rename from Unix -> Posix and adding these private ctor/dtor deletions, was anything else changed that is worth a review?

Reply:
No.

Reply:
* I rewritten `DateTimeToPosixTime()` to not use temporary variable.
* Doxy comment updates (nothing big, just touchups)